src/components/customers/subscriptions/AddPlanToCustomerDialog.tsx

```diff
@@ -0,0 +1,216 @@
+import { forwardRef, useMemo, RefObject, useState, useImperativeHandle, useRef } from 'react'
+import { gql } from '@apollo/client'
+import styled from 'styled-components'
+import { useFormik } from 'formik'
+import { object, string } from 'yup'
+
+import { Dialog, Button, DialogRef, Alert, Typography } from '~/components/designSystem'
+import { ComboBoxField, TextInputField } from '~/components/form'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { addToast, LagoGQLError } from '~/core/apolloClient'
+import { theme } from '~/styles'
+import {
+  useCreateSubscriptionMutation,
+  CustomerSubscriptionListFragmentDoc,
+  useGetPlansLazyQuery,
+  CreateSubscriptionInput,
+  Lago_Api_Error,
+} from '~/generated/graphql'
+
+export interface AddPlanToCustomerDialogRef {
+  openDialog: (existingInfos?: { subscriptionId: string; existingPlanId: string }) => unknown
+  closeDialog: () => unknown
+}
+
+gql`
+  query getPlans($page: Int, $limit: Int) {
+    plans(page: $page, limit: $limit) {
+      collection {
+        id
+        name
+        code
+      }
+    }
+  }
+
+  mutation createSubscription($input: CreateSubscriptionInput!) {
+    createSubscription(input: $input) {
+      ...CustomerSubscriptionList
+    }
+  }
+
+  ${CustomerSubscriptionListFragmentDoc}
+`
+
+interface AddPlanToCustomerDialogProps {
+  customerName: string
+  customerId: string
+  refetchCustomer: () => void
+}
+
+export const AddPlanToCustomerDialog = forwardRef<
+  AddPlanToCustomerDialogRef,
+  AddPlanToCustomerDialogProps
+>(({ customerId, customerName, refetchCustomer }: AddPlanToCustomerDialogProps, ref) => {
+  const dialogRef = useRef<DialogRef>(null)
+  const [existingInfos, setExistingInfos] = useState<
+    | {
+        subscriptionId: string
+        existingPlanId: string
+      }
+    | undefined
+  >(undefined)
+  const { translate } = useInternationalization()
+  const [getPlans, { loading, data }] = useGetPlansLazyQuery()
+  const [create] = useCreateSubscriptionMutation({
+    context: {
+      silentErrorCodes: [Lago_Api_Error.CurrenciesDoesNotMatch],
+    },
+    onCompleted: async ({ createSubscription }) => {
+      if (!!createSubscription) {
+        await refetchCustomer()
```

Comment:
Should onComplete hook handle fetch ? I thought the refetchQueries in onSubmit was supposed to handle such

Reply:
Anyway, not blocking

Reply:
Both works but I'll change it for more concistency